java/src/com/ibm/streamsx/rest/RESTException.java

```diff
@@ -25,28 +27,39 @@
      * @param code
      *            - error message code (matches HTTP response codes)
      *
-     * @return a {@link RESTException} created from a code and an IBM Streams Message
+     * @return a {@link RESTException} created from a code and an IBM Streams
+     *         Message
      */
     public static final RESTException create(int code, String streamsMessage) {
-        RESTErrorMessage error = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create().fromJson(streamsMessage,
-                RESTErrorMessage.class);
-        return new RESTException(code, error);
+        RESTErrorMessage error = null;
+        RESTException rcException;
+        try {
+            error = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create().fromJson(streamsMessage,
+                    RESTErrorMessage.class);
+        } catch (IllegalStateException e) {
+            // chances are this is a 404 http error with no streams message but
+            // something else
+            rcException = new RESTException(code, streamsMessage);
```

Comment:
yes, you are correct.  line 44 was suppose to be inside the try block.  i've amended the commit for this.